File: frontend/admin/src/api/usage.ts
import api from './index';

export interface UsageStats {
  total_input_tokens: number;
  total_output_tokens: number;
  total_tokens: number;
  total_search_calls: number;
  total_api_calls: number;
}

export interface UserUsageItem extends UsageStats {
  user_id: string;
  username?: string;
}

export interface AllUsageResponse {
  users: UserUsageItem[];
  summary: UsageStats & { total_user_count: number };
}

export interface DailyUsageItem {
  date: string;
  input_tokens: number;
  output_tokens: number;
  search_calls: number;
}

export async function fetchMyUsage(period: string = 'all'): Promise<UsageStats> {
  const res = await api.get(`/api/usage/my?period=${period}`);
  return res.data;
}

export async function fetchAllUsage(period: string = 'all'): Promise<AllUsageResponse> {
  const res = await api.get(`/api/usage/all?period=${period}`);
  return res.data;
}

export async function fetchMyDailyUsage(days: number = 30): Promise<DailyUsageItem[]> {
  const res = await api.get(`/api/usage/my/daily?days=${days}`);
  return res.data;
}

export async function fetchAllDailyUsage(days: number = 30): Promise<DailyUsageItem[]> {
  const res = await api.get(`/api/usage/all/daily?days=${days}`);
  return res.data;
}

File: backend/routers/usage.py
from fastapi import APIRouter, Depends, Query
from pydantic import BaseModel
from typing import Optional
from routers.deps import get_current_user, require_super_admin
from services.usage_service import usage_service
from services.models.user import User

router = APIRouter(prefix="/api/usage", tags=["usage"])


class UsageResponse(BaseModel):
    total_input_tokens: int
    total_output_tokens: int
    total_tokens: int
    total_search_calls: int
    total_api_calls: int


class UserUsageItem(BaseModel):
    user_id: str
    username: str = ""
    total_input_tokens: int
    total_output_tokens: int
    total_tokens: int
    total_search_calls: int
    total_api_calls: int


class AllUsageResponse(BaseModel):
    users: list[UserUsageItem]
    summary: dict


class DailyUsageItem(BaseModel):
    date: str
    input_tokens: int
    output_tokens: int
    search_calls: int


@router.get("/my", response_model=UsageResponse)
async def get_my_usage(
    period: str = Query("all", regex="^(today|7d|30d|all)$"),
    current_user: User = Depends(get_current_user),
):
    return usage_service.get_user_usage(current_user.id, period)


@router.get("/all", response_model=AllUsageResponse)
async def get_all_users_usage(
    period: str = Query("all", regex="^(today|7d|30d|all)$"),
    current_user: User = Depends(require_super_admin),
):
    return usage_service.get_all_users_usage(period)


@router.get("/my/daily", response_model=list[DailyUsageItem])
async def get_my_daily_usage(
    days: int = Query(30, ge=1, le=365),
    current_user: User = Depends(get_current_user),
):
    return usage_service.get_daily_breakdown(current_user.id, days)


@router.get("/all/daily", response_model=list[DailyUsageItem])
async def get_all_daily_usage(
    days: int = Query(30, ge=1, le=365),
    current_user: User = Depends(require_super_admin),
):
    return usage_service.get_daily_breakdown(days=days)
